### PR TITLE
[RQ-827] feat: added banner to show feature limit reached

### PR DESCRIPTION
**Pull request by nafees87n**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/components/features/rules/RulesListContainer/index.js [hunk 1]
```diff
@@ -231,7 +232,7 @@ const RulesListContainer = ({ isTableLoading = false }) => {
   return (
     <>
       {/* Page content */}
-
+      {user.details.isLimitReached ? <FeatureLimiterBanner /> : null}
```
Comment (RuntimeTerror10) [inline app/src/components/features/rules/RulesListContainer/index.js:235]: Add a check here for user is logged in or not, might break for users have rules in logged out state

File: app/src/hooks/featureLimiter/featureLimitTypes.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export const featureLimits: PlanFeatureLimits = {
+  [APP_CONSTANTS.PRICING.PLAN_NAMES.FREE]: {
+    [FeatureLimitType.num_rules]: 10,
+    [FeatureLimitType.num_active_rules]: 3,
+  },
+  [APP_CONSTANTS.PRICING.PLAN_NAMES.BASIC]: {
+    [FeatureLimitType.num_rules]: 25,
+    [FeatureLimitType.num_active_rules]: 10,
+  },
+};
```
Comment (RuntimeTerror10) [inline app/src/hooks/featureLimiter/featureLimitTypes.ts:21]: Lets only maintain types in this file. Can we move this to a new `constants.ts` file?

File: app/src/hooks/featureLimiter/useFeatureLimiter.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  }, [dispatch, isLimitReached]);
+
+  const checkFeatureLimits = () => {
+    if (!isUserLoggedIn) {
+      return;
+    }
+
+    if (isUserPremium && userPlan !== APP_CONSTANTS.PRICING.PLAN_NAMES.BASIC) {
+      return;
+    }
```
Comment (RuntimeTerror10) [inline app/src/hooks/featureLimiter/useFeatureLimiter.ts:33]: Why do we need specific check for basic plan here?

File: app/src/hooks/featureLimiter/featureLimitTypes.ts (hunk 2 above)
Comment (nafees87n) [reply app/src/hooks/featureLimiter/featureLimitTypes.ts:21]: Moved the types to `types.ts`

**Review by RuntimeTerror10 — APPROVED** [review]

**Pull request merged**